release-calendar/src/server/repository-service.ts

```diff
@@ -44,21 +54,16 @@ export class RepositoryService {
     return releaseQuery;
   }
 
-  getReleases(): Promise<Release[]> {
-    const releaseQuery = this.releaseRepository
-      .createQueryBuilder('release')
-      .leftJoinAndSelect('release.promotions', 'promotion')
-      .getMany();
-
-    releaseQuery.then((releases) => {
-      releases.forEach((release) =>
-        release.promotions.sort(
-          (a, b): number => b.date.getTime() - a.date.getTime(),
-        ),
-      );
-    });
-
-    return releaseQuery;
+  getReleases(): Promise<Promotion[][]> {
```

Comment:
What would be a sample output for this method?

Reply:
[[LTS promotion, LTS promotion,..], [Stable promotion, Stable promotion,..], [Opt-in Bets promotion, Opt-in Beta promotion,..],...]

Reply:
Hmm... can you make this a `Map<Channel, Promotion[]>` instead?
```typescript
{
  Channel.LTS: [LTS Promotion, LTS Promotion, ...],
  Channel.Stable: [Stable promotion, Stable promotion, ...],
  Channel.OptInBeta: [Opt-in Beta promotion, Opt-in Beta promotion, ...],
  ...
}
```